src/main/java/core/basesyntax/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.dao.Storage;
+import core.basesyntax.dto.Transaction;
+import core.basesyntax.services.Validator;
+import core.basesyntax.services.impl.ValidatorImpl;
+
+public class PurchaseOperation implements Operation {
+    @Override
+    public int apply(Transaction transaction) {
+        Validator validator = new ValidatorImpl();
+        Storage.map.computeIfAbsent(transaction.getFruit().getFruitName(),
+                k -> transaction.getQuantity());
+        int oldQuantity = Storage.map.get(transaction.getFruit().getFruitName());
+        int newQuantity = oldQuantity - transaction.getQuantity();
+        if (validator.checkOperation(newQuantity)) {
+            Storage.map.put(transaction.getFruit().getFruitName(), newQuantity);
+        }
+        return newQuantity;

```

Comment:
do it if you want
you can make DAO class that will have get() and put() methods that will work with database, and use them here. It will bring closer your code to real projects, and first of all make more readable